### DISCUSSION TITLE
GNS3 with AdvantEdge

### DISCUSSION THREAD

**Discussion by anjumm** [InterDigitalInc/AdvantEDGE#426] [General]
Hi, 

Hope you all are doing well. 
I am currently working on a setup in which I would like to connect GNS3 with AdvantEDGE. The main idea is to use the mobility feature of AdvantEDGE. Additionally using network and edge servers from GNS3. I tried to look for documentation related to integration of both applications but couldn't find much information. I will appreciate if I get some guidance regarding it. 

I did find External Node Support information, so is it similar for GNS3 integration? 
https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/ext-nodes/

Thanks in advance for the support.

**Comment by dilallkx** [comment]
Hi @anjumm,

Thank you for reaching out, and apologies for the delayed response!

If I understand correctly, you would like to set up your network topology and run your applications using GNS3 and then trigger mobility events using AdvantEDGE to dynamically change/modify the GNS3 topology. Is this correct? If so, I believe GNS3 and AdvantEDGE are conflicting solutions; i.e. you can also use AdvantEDGE to create a network scenario/topology to emulate end-to-end network characteristics for the applications. You can then trigger mobility events to change a UE’s POA, which will trigger a recalculation and enforcement of new network characteristics for application traffic. The mobility events are used to dynamically change the deployed AdvantEDGE network scenario; these events cannot be used to change the GNS3 network topology.

Depending on your testing needs, you could also consider using AdvantEDGE to test your applications instead of GNS3. Your applications could be deployed in a Linux container environment or as external applications that simply pass application traffic through the AdvantEDGE platform. In this case you would benefit from the platform capabilities such as end-to-end network characteristics emulation (latency, jitter, packet loss, throughput) and dynamic event injection (e.g. mobility events).

I hope this clarifies what AdvantEDGE can do for you. Please let me know if you have additional questions or details about your use case that I could help with.

Best regards!
